### PR TITLE
Don't run the E2E tests on installer documentation

### PR DESCRIPTION
**Pull request by crawford**

### CODE REVIEW

File: ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml [hunk 1]
```diff
@@ -35,6 +35,7 @@ presubmits:
     - dockerless-build-temporary
     rerun_command: "/test e2e-aws"
     always_run: true
+    run_if_changed: "^((?!Documentation).)*$"
```
Comment (wking) [inline ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml]: Maybe just:

```yaml
run_if_changed: '^(Documentation/.*)
```

?

File: ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml [hunk 2]
```diff
@@ -35,6 +35,8 @@ presubmits:
     - dockerless-build-temporary
     rerun_command: "/test e2e-aws"
     always_run: true
```
Comment (wking) [inline ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml]: You need to drop this to [avoid][1]:

```
# test that the prow config is parseable
time="2018-08-14T23:10:51Z" level=fatal msg="Error loading config." error="job pull-ci-origin-installer-e2e-aws is set to always run but also declares run_if_changed targets, which are mutually exclusive" 
```

[1]: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_release/1191/pull-ci-openshift-release-config/285/build-log.txt

Comment (crawford) [reply ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml]: I was wondering about that. The documentation for prow features an example that does exactly this.

File: ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml [hunk 3]
```diff
@@ -34,7 +34,7 @@ presubmits:
     - master
     - dockerless-build-temporary
     rerun_command: "/test e2e-aws"
-    always_run: true
+    run_if_changed: "^([^D]|D(D|oD|ocD|ocuD|ocum(D|e(D|n(D|t(D|aD|atD|atiD|atioD)))))*([^Do]|o[^Dc]|oc[^Du]|ocu[^Dm]|ocum([^De]|e([^Dn]|n([^Dt]|t([^Da]|a[^Dt]|at[^Di]|ati[^Do]|atio[^Dn]))))))*(D(D|oD|ocD|ocuD|ocum(D|e(D|n(D|t(D|aD|atD|atiD|atioD)))))*(o|oc|ocu|ocum(e(n(t(a|at|ati|atio)?)?)?)?)?)?$"
```
Comment (yifan-gu) [inline ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml]: nit: Is it possible to add a comment of this?

File: ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml [hunk 4]
```diff
@@ -34,7 +34,9 @@ presubmits:
     - master
     - dockerless-build-temporary
     rerun_command: "/test e2e-aws"
-    always_run: true
+    # The abomination below is equivilent to `^((?!Documentation).)*$`. Since
+    # Go doesn't support negative lookaheads, we are stuck with the following.
+    run_if_changed: "^([^D]|D(D|oD|ocD|ocuD|ocum(D|e(D|n(D|t(D|aD|atD|atiD|atioD)))))*([^Do]|o[^Dc]|oc[^Du]|ocu[^Dm]|ocum([^De]|e([^Dn]|n([^Dt]|t([^Da]|a[^Dt]|at[^Di]|ati[^Do]|atio[^Dn]))))))*(D(D|oD|ocD|ocuD|ocum(D|e(D|n(D|t(D|aD|atD|atiD|atioD)))))*(o|oc|ocu|ocum(e(n(t(a|at|ati|atio)?)?)?)?)?)?$"
```
Comment (stevekuznetsov) [inline ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml]: How often does your repository change top-level directories? Seems like you could have called those out in a much more simple manner.

Comment (wking) [reply ci-operator/jobs/openshift/installer/openshift-installer-presubmits.yaml]: > How often does your repository change top-level directories?

We have a top-level change in flight now with openshift/release#120 ;).  But we'll need CI changes in other places to adapt to that anyway, so I wouldn't mind switching to a whitelist approach here too.  I'm also fine with the blacklist approach.

**Pull request merged**